src/main/java/com/guciowons/shoppingbasket/Basket/Summary.java

```diff
@@ -0,0 +1,62 @@
+package com.guciowons.shoppingbasket.Basket;
+
+import com.guciowons.shoppingbasket.Product.Product;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class Summary {
```

Comment:
Tutaj znowu łączysz dwie odpowiedzialności -- trzymania reprezentacji danych (np `BasketSummarized`) i przeliczenia koszyka (np `BasketSummarizer`), to powinny być dwie osobne klasy.

`BasketSummarizer` (albo inna nazwa) powinien dostać stan koszyka, podsumować go i zwrócić zwykłe POJO, które będzie używane na warstwie RESTa tylko do zwrócenia wyniku.

Jeśli robisz to w ten sposób to tylko gettery i settery sterują tym, jak Jackson (spring-web) ma zserializować ten obiekt do wyświetlanej formy.